controllers/useraccount/useraccount_controller.go

```diff
@@ -484,6 +492,29 @@ func (r *Reconciler) deleteUser(ctx context.Context, userAcc *toolchainv1alpha1.
 	return true, nil
 }
 
+// deleteUserResources deletes the user settings resources (configmap, role and role-binding) associated with the specified user, created for a user by console.
+// This function only looks for these resources in the namespace - openshift-console-user-settings
+// Returns an error if any of the deletion operations fail.
+func (r *Reconciler) deleteUserResources(ctx context.Context, userUID string) error {
+
+	// Users which were created in the cluster with the OCP versions which includes https://issues.redhat.com/browse/OCPBUGS-32321 fix
+	// will have a label which will help to map the User to the User settings resources.
+
+	// Users created before that won't have that label, and we have to rely on the name of the resource being of type `user-settings-<UserUID>`,
+	// delete ConfigMap, Role and RoleBinding
+
+	if err := deleteResource(ctx, r.Client, userUID, &corev1.ConfigMap{}); err != nil {
+		return err
+	}
+	if err := deleteResource(ctx, r.Client, userUID, &rbac.Role{}); err != nil {
+		return err
+	}
+	if err := deleteResource(ctx, r.Client, userUID, &rbac.RoleBinding{}); err != nil {
+		return err
+	}
+	return nil
```

Comment:
just a detail
```suggestion
	return deleteResource(ctx, r.Client, userUID, &rbac.RoleBinding{})
```

Reply:
done in [7b7bd25](https://github.com/codeready-toolchain/member-operator/pull/568/commits/7b7bd259bb5cb92d555a5043585a0d3c06875e8b)